QueueInteropTransport.php

```diff
@@ -221,21 +218,61 @@ private function setMessageMetadata(Message $interopMessage, Envelope $envelope)
         }
     }
 
-    private function encodeMessage(Envelope $envelope): Message
+    private function encodeMessage(Envelope $envelope): InteropMessage
     {
         $context = $this->contextManager->context();
         $encodedMessage = $this->serializer->encode($envelope);
 
+        if ($context instanceof AmqpContext
+            && !$this->serializer instanceof PhpSerializer
```

Comment:
What if someone is using .a different serializer?

I dont really understand why we should special treat AMQP context and PHPSerializer. Could you please explain that?